index.js

```diff
@@ -147,3 +159,44 @@ module.exports = {
     return entityName;
   },
 };
+
+/**
+  * Custom info derived from CLI options for use within this blueprint.
+  * Nothing in this object is expected from the blueprint system.
+  */
+function addonInfoFromOptions(options) {
+  let addonEntity = options.entity;
+  let addonRawName = addonEntity.name;
+
+  return {
+    name: {
+      dashed: stringUtil.dasherize(addonRawName),
+      classified: stringUtil.classify(addonRawName),
+      raw: addonRawName,
+    },
+    entity: addonEntity,
+    location: options.addonLocation || path.join('packages', addonRawName),
```

Comment:
Wouldn't this need to be the dasherized addon name?

```suggestion
    location: options.addonLocation || path.join('packages', stringUtil.dasherize(addonRawName)),
```

Reply:
I don't think so. Why would it need to be?

Reply:
It was so before, and I think this is also what ember-cli for v1 addons does. For example when the user entered a name including spaces, we will probably not want to have spaces in the path?

Reply:
Spaces aren't allowed for package names anyway. We should throw in that case

For, doing what v1 addons do, i don't think we should copy those blindly. We have a chance to really look at the behavior of the blueprint, so we should take it and discuss -- try to make things less magical, more straightforward, etc

Reply:
Well, but we currently do the same magic as the v1 blueprint, e.g. turn a name with spaces into a dasherized version for the package.json name. So we should do the same for the path IMO.

> For, doing what v1 addons do, i don't think we should copy those blindly. We have a chance to really look at the behavior of the blueprint, so we should take it and discuss -- try to make things less magical, more straightforward, etc

Yeah, I see your point. But if we want to do that, then this new behavior should be consistent, see my example above. And it should be a separate PR I think.

Reply:
aight, fair enough -- pushed! :partying_face: